SingularityUI/app/components/globalSearch/GlobalSearch.cjsx

```diff
@@ -0,0 +1,85 @@
+React = require 'react'
+classNames = require 'classnames'
+
+{ Typeahead } = require 'react-typeahead'
+fuzzy = require 'fuzzy'
+
+class GlobalSearch extends React.Component
+  optionSelected: (requestIdObject) =>
+    requestId = @getValueFromOption(requestIdObject)
+    app.router.navigate "/request/#{ requestId }", { trigger: true }
+    @clear()
+    @props.onHide()
+
+  resetSelection: =>
+    @refs.typeahead.setState({
+      selectionIndex: 0
+    })
+
+  clear: =>
+    @refs.typeahead.setEntryText('')
+    @resetSelection()
+
+  focus: =>
+    @refs.typeahead.focus()
+    @resetSelection()
+
+  searchOptions: (inputValue, options) ->
+    # fuzzy lazily just appends a string before and after a matching char
+    # we have to later use a simple shift-in shift-out state machine to convert
+    fuzzyOptions = {
+      returnMatchInfo: true
+    }
+
+    searched = fuzzy.filter(inputValue, options, fuzzyOptions)
+
+    return searched
+
+  renderOption: (option, index) ->
+    # transform fuzzy string into react component
+    bolded = option.string.map((matchInfo) ->
+      if matchInfo.match then <b>{matchInfo.char}</b> else <span>{matchInfo.char}</span>
+    )
+
+    return bolded
+
+  getValueFromOption: (option) ->
+    return option.original
+
+  render: =>
+    if @props.visible
+      @focus()
```

Comment:
the more idiomatic way to do this would be to use [componentDidUpdate()](https://facebook.github.io/react/docs/component-specs.html#updating-componentdidupdate), and only call `@focus()` if the `visible` prop transitioned from `false` to `true` (as a general rule of thumb, you shouldn't be manipulating the DOM or triggering events like that in `render()`)